### ISSUE TITLE
Switch to `fastembed`

### ISSUE THREAD

**Issue by nleroy917** [pepkit/pephub#230]
[`fastembed`](https://github.com/qdrant/fastembed) is a new package that enables really lightweight, inference on a lot of pre-trained natural language models. Notably, it supports the [all-MiniLM-L6-v2](https://qdrant.github.io/fastembed/examples/Supported_Models/) model we use on pephub.

It seems like it is both faster and less bloated (see https://github.com/qdrant/fastembed/pull/13):
>  Huggingface Transformers (Average, Max, Min): (0.1034623384475708, 0.10434174537658691, 0.10258293151855469)
FastEmbed (Average, Max, Min): (0.06690812110900879, 0.06795573234558105, 0.06586050987243652)
>
>and my venv directory is now only 200Mb instead of 5Gb for optimum

I think we would get a lot of benefits from using this over vanilla `sentence-transformers`. There are four steps to achieve this:

- [x] Add `fastembed` to `requirements.txt`
- [x] Re-embed our database using `fastembed`, store in qdrant.
- [x] Remove `SentenceTransformers` from `/search` endpoints and replace with `fastembed`'s `FlagEmbedding` class. See their [docs](https://github.com/qdrant/fastembed#-usage)
- [x] Remove `sentence-transformers` dependency.

**Comment by nleroy917** [comment]
FYI the [MTEB leaderboad](https://huggingface.co/spaces/mteb/leaderboard) shows [bge-small-en-v1.5](https://huggingface.co/BAAI/bge-small-en-v1.5) as being much better for retrieval tasks - maybe we switch models?

**Comment by nleroy917** [comment]
We should probably use the `bge-small-en-v1.5` since there are known issues with `all-MiniLM-L6-v2` over in `fastembed`: https://github.com/qdrant/fastembed/issues/22

**Comment by nleroy917** [comment]
I've switched over to `fastembed`, I just need to reindex the actual database

**Comment by nleroy917** [comment]
Database is reindexed which will technically break pephub search in prod - but we can redeploy the current server and change the env variable for the embedding model which will fix it.

**Comment by nleroy917** [comment]
After this, the container is still a bit large, but better (1.1G on my machine). Top offenders are:
- the frontend (node_modules) Can we make this smaller?
```
65M     typescript
37M     @swc
26M     handsontable
15M     core-js
12M     hyperformula
11M     prettier
9.8M    bootstrap-icons
9.8M    bootstrap
9.3M    @typescript-eslint
9.1M    @esbuild
7.9M    es-abstract
7.9M    @tanstack
7.8M    @babel
5.2M    moment
5.1M    sass
5.1M    @types
5.0M    lodash
4.5M    react-bootstrap
4.4M    react-dom
4.2M    eslint
3.3M    vite
3.2M    framer-motion
2.5M    rollup
2.5M    @emotion
2.4M    @popperjs
2.3M    @remix-run
2.2M    axe-core
2.2M    @restart
2.1M    axios
2.0M    react-hook-form
```
- some python packages
```
# du -sh * | sort -r -h
80M     pandas
65M     onnx
52M     sympy
31M     numpy
27M     numpy.libs
19M     sqlalchemy
19M     onnxruntime
15M     pip
14M     tokenizers
14M     cryptography
13M     grpc
8.1M    pygments
8.0M    psycopg2_binary.libs
7.1M    grpc_tools
5.2M    rpds
3.9M    mpmath
3.4M    setuptools
2.8M    tzdata
2.8M    pytz
2.7M    greenlet
2.6M    yaml
2.4M    qdrant_client
2.4M    huggingface_hub
2.3M    rich
2.3M    _pytest
1.8M    google
1.7M    pkg_resources
1.4M    fsspec
1.2M    fastapi
1.1M    pycparser
1.1M    _cffi_backend.cpython-310-aarch64-linux-gnu.so
992K    jinja2
```

**Comment by khoroshevskyi** [comment]
This issue was solved in v0.11.0
